python/pyspark/sql/types.py

```diff
@@ -747,8 +747,8 @@ def jsonValue(self) -> Dict[str, Any]:
     def fromJson(
         cls,
         json: Dict[str, Any],
-        fieldPath: str,
-        collationsMap: Optional[Dict[str, str]],
+        fieldPath: str = "",
+        collationsMap: Optional[Dict[str, str]] = None,
     ) -> "ArrayType":
         elementType = _parse_datatype_json_value(
             json["elementType"], fieldPath + ".element", collationsMap
```

Comment:
Seems like it's `.element` when empty strings are given. Are they expected?

Reply:
Yes, if you are deserializing a schema where an array is on the column `a` the `collationsMap` would be `{"a.element": collation}`. So if you just want to deserialize a map that is not a part of the larger schema the `collationsMap` should be `{".element": collation}`

Added tests to verify this.

Reply:
I feel like it has to be `{"element": collation}` and `{"key": collation}`. `.element` sounds a bit awkward. While I am fine because this JSON format is supposed to be internal-only, is this backward compatible? E.g., can old JSON data types be read? I have seen many tickets related to this, e.g., https://github.com/apache/spark/pull/43474